calculator.rb

```diff
@@ -0,0 +1,139 @@
+# ROOM FOR IMPROVEMENT
+# Support parentheticals
+# Support commas in longer numbers
+# Allow for equation with more than 2 numbers
+# Support square root
+# Ask whether to end or continue after outputting result
+
+# -------------
+
+# CHECKS IF USER INPUT IS A VALID OPERATION
+def is_operator?(user_input)
+  operators = ["+", "add", "addition", "plus",
+               "-", "sub", "subtract", "subtraction",
+               "*", "multiply", "times",
+               "/", "divide", "division",
+               "^", "**", "power", "exponent", "exponents", "raise",
+               "%", "modulo", "modulus", "remainder"
+  ]
+
+  if operators.include?(user_input)
+    return user_input
+  else
+    while !(operators.include?(user_input))
+      # conditional informs user of the type of error
+      if user_input == "" # nil
+        puts "\nUm… you have to enter *something*, bud."
+      else
+        # if user_input is a word or unknown symbol
+        puts "\nHey, now. We don't do that kind of math here."
+      end
+
+      # prompts user for valid operator
+      print "Let's try that again. What operator would you like to use? "
+      user_input = gets.chomp
```

Comment:
Interesting, your `while` loop doesn't have an explicit return value, but instead taking advantage of a feature in Ruby where functions return the value of the last executed statement (in this case, the `gets.chomp`)